docs/architecture.md

```diff
@@ -112,71 +112,199 @@ The redux store is composed of two `slices` : `documents` and `queries`:
+When doing so, the app loads the store when it starts and benefit from the instant available cached data.
+
+But, the data could have significantly evolved between the last store save and the start, making the documents potentially stale. 
+Hopefully, since the fetch policy is probably expired, queries will be run again, retrieving fresh data. Unfortunately, this might cause an immediate spinning display on the app when it starts, even though there is available data thanks to the persisted store. 
+
+To solve this, a `backgroundFetching` option can be set, at the query level, or at the store level (enabled for all queries).
+
+When the query is loaded, it simply keeps the query status in `loaded` state (rather than setting a `loading` state) and set a new `isFetching` attribute, stored in the query. 
+Thanks to this, the app can adapt its display to avoid the "spinning effect", and might inform the user that data is being updated in the background, without refreshing the whole app.
+
+See [this PR](https://github.com/cozy/cozy-client/pull/1211) for more insights.
+
+
+Here is a simplified execution sequence with or without background fetching:
+```mermaid
+---
+title: Background fetching
+---
+sequenceDiagram
+    participant App as App
+    participant CC as cozy-client
+    participant Store as Store
+    participant Backend as Backend
+
+    App->>CC: Query documents
+
+    CC->>Store:  Load query
+    alt Without background fetching
+      Store->>Store: update query status
+      Store->>App: Loading query status
+      App->>App: Spinner
+    else With background fetching
+      Store->>Store: no change on query status
+      Store->>App: isFetching status
+      App->>App: Do nothing
+    end
+    CC->>Backend:  Execute query
+    Backend->>CC:  New data
+    CC->>Store: Update store
+    CC->>App: New data
+    App->>App: Refresh display
+```
+
+### Updates
+
+Here, we detail what happens when a document is updated, i.e. a mutation happens, typically after a `client.save`.
+
+If the document `_id` does not exist in the store's `documents` collection, i.e. after the a creation, it is simply added.
+If it already exists, the document content is updated in this collection.
+
+Furthermore, just like with the queries, any mutation triggers the [auto query updater](#auto-query-update), that can result in adding or removing document ids.
+
+Thanks to it, the app is guaranteed to have the up-to-date documents, returned from the `documents` collection.
+
+⚠️ As already mentioned, when a mutated document already exists in the store throught its `_id`, it is updated with the new content. However, this update never removes existing fields, it only merges or adds new fields. This is typically useful when two queries retrieve the same document, but not with all the fields. 
```

Comment:
I think this based on the attribute `_rev` and not `_id` ([code](https://github.com/cozy/cozy-client/blob/ea8b7909ac8241fcc443fc16536985b702966e40/packages/cozy-client/src/CozyClient.js#L782))

Reply:
Fair enough, but the code you mention is used to determine rather it is a create or update mutation, that will be used afterwards by the Link to call a create or update on the collection: https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/StackLink.js#L117-L119
Here, we are on the documents store perspective, which rely on the `_id` existence:  https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/store/documents.js#L178-L183